### ISSUE TITLE
[Question] Can this be used broadly for all microeukaryotes (e.g., fungi, protists, etc)?

### ISSUE THREAD

**Issue by jolespin** [weberlab-hhu/Helixer#100]
I'm thinking about trying this out for a backend for my VEBA eukaryotic binning module (https://github.com/jolespin/veba) as an alternative to MetaEuk.

Looking at the examples here: 
```
Helixer.py --lineage land_plant --fasta-path Arabidopsis_lyrata.v.1.0.dna.chromosome.8.fa  \
  --species Arabidopsis_lyrata --gff-output-path Arabidopsis_lyrata_chromosome8_helixer.gff3
```

I won't have the lineage or species at this point in the pipeline.  

* How important is the `--lineage` parameter in the gene calling process?
* Can `--species` be anything? For example, can I give it a bin ID such as `S1__METABAT2__bin.1`? 
* Will this be available on bioconda any time soon?

**Comment by BjoernUsadel** [comment]
Hi 
the lineage selects the trained models currently one of 

- land plants
- invertebrates
- vertebrates
- fungi
hence you need this or alternatively model filepath set with "--model-filepath"

Do you want to align it to the BUSCO lineages? This could be done by a simple lookup. 
However as gene paramters change for the gene detection depending on these broadf lineages these are needed.
Cheers
b

**Comment by jolespin** [comment]
I meant if Helixer had a model for each eukaryotic BUSCO lineage, then it does auto detection in the backend to know which model to use.

**Comment by alisandra** [comment]
Hi Jolespin, 

Thanks for your interest!

This is a cool idea on auto-detection. One would have to test it, but I'd hazard a guess that auto detection might work off of the confidence of Helixer's raw predictions. 

That said, I don't currently have much capacity for larger schemes, so in summary
- yes, `--lineage` is critical for current models, else a poorly fitting model would be used; and prediction quality would suffer massively.
- `--species` should be any reasonable length non-white space string. It's going to end up in the gff3 file's gene names. A prefix such as 'At' or 'A_thal' for Arabidopsis thaliana would be reasonable. The bin ID should work without issue. 
- not in the immediate future
- not in the immediate future

That said on the no's, if any one wants to see this enough to give it a shot, I'm happy to do what I can at a high-level; drop a line.

**Comment by jolespin** [comment]
Understood the bandwidth issue.  I'm in the same boat right now w/ my VEBA package. 

I would be using this for microeukaryotic organisms (e.g., protists).  Do you have any instructions on training a custom model? If so, what is required for this? Genomic and CDS sequences or could this be done directly from protein sequences?

**Comment by alisandra** [comment]
Yes, I'm currently working on the latest instructions here: https://github.com/weberlab-hhu/Helixer/blob/cleanup/docs/training.md
(I will merge them to `main` soon).

You will need fasta and gff3 files for the training species (it's supervised training). I am not on top of the availability of 
good references in protists, I can imagine it might be a challenge. 

Drop questions whenever!

**Comment by jolespin** [comment]
For the training data, are genes with alternative start codons used or discarded?

**Comment by alisandra** [comment]
In the current default implementation, genes with non-ATG start codons are used, but the upstream region is masked. So the network will learn there's a gene there, but not receive feedback on where exactly it started. 

This behavior is of course not designed for alternative start codons, but is a side effect of the partial-gene-model detection and masking, which assumes the standard genetic code. 

In general, supporting genetic code variants is on the "think about it" list; not doing so yet has been a rare--but still noticeable--issue in fungi already.

**Comment by jolespin** [comment]
Any updates on whether or not this should be able to handle protists soon?
